lib/resty/redis.lua

```diff
@@ -82,11 +86,51 @@ function _M.connect(self, ...)
         return nil, "not initialized"
     end
 
+    local params = {...}
```

Comment:
we should avoid creating a new table here for better performance, by using `_M.connect(self, host, port_or_opts, opts)`

Reply:
refactored accordingly